File: Samples/CardsStarterKit_4_0_VB/CardsStarterKit_4_0/CardsGame/CardsGame/Screens/MainMenuScreen.vb
#Region "File Description"
'-----------------------------------------------------------------------------
' MainMenuScreen.vb
'
' Microsoft XNA Community Game Platform
' Copyright (C) Microsoft Corporation. All rights reserved.
'-----------------------------------------------------------------------------
#End Region

Imports System.Text
Imports Blackjack.GameStateManagement


Friend Class MainMenuScreen
    Inherits MenuScreen
    Public Shared Theme As String = "Red"

#Region "Initializations"

    ''' <summary>
    ''' Initializes a new instance of the screen.
    ''' </summary>
    Public Sub New()
        MyBase.New("")
    End Sub
#End Region

    Public Overrides Sub LoadContent()
        ' Create our menu entries.
        Dim startGameMenuEntry As New MenuEntry("Play")
        Dim themeGameMenuEntry As New MenuEntry("Theme")
        Dim exitMenuEntry As New MenuEntry("Exit")

        ' Hook up menu event handlers.
        AddHandler startGameMenuEntry.Selected, AddressOf StartGameMenuEntrySelected
        AddHandler themeGameMenuEntry.Selected, AddressOf ThemeGameMenuEntrySelected
        AddHandler exitMenuEntry.Selected, AddressOf OnCancel

        ' Add entries to the menu.
        MenuEntries.Add(startGameMenuEntry)
        MenuEntries.Add(themeGameMenuEntry)
        MenuEntries.Add(exitMenuEntry)

        MyBase.LoadContent()
    End Sub

#Region "Update"
    ''' <summary>
    ''' Respond to "Play" Item Selection
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub StartGameMenuEntrySelected(ByVal sender As Object, ByVal e As EventArgs)
        For Each screen In ScreenManager.GetScreens
            screen.ExitScreen()
        Next screen

        ScreenManager.AddScreen(New GameplayScreen(Theme), Nothing)
    End Sub

    ''' <summary>
    ''' Respond to "Theme" Item Selection
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub ThemeGameMenuEntrySelected(ByVal sender As Object, ByVal e As EventArgs)
        ScreenManager.AddScreen(New OptionsMenu, Nothing)
    End Sub

    ''' <summary>
    ''' Respond to "Exit" Item Selection
    ''' </summary>
    ''' <param name="playerIndex"></param>
    Protected Overrides Sub OnCancel(ByVal playerIndex As PlayerIndex)
        ScreenManager.Game.Exit()
    End Sub
#End Region
End Class

File: Samples/CatapultWars_4_0/Source/EX2_PolishAndMenus/Start/CatapultGame/CatapultGame/Screens/MainMenuScreen.cs
#region File Description
//-----------------------------------------------------------------------------
// MainMenuScreen.cs
//
// Microsoft XNA Community Game Platform
// Copyright (C) Microsoft Corporation. All rights reserved.
//-----------------------------------------------------------------------------
#endregion

#region Using Statements
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using GameStateManagement;
using Microsoft.Xna.Framework;
#endregion

namespace CatapultGame
{
    class MainMenuScreen : MenuScreen
    {
        public MainMenuScreen()
            : base(String.Empty)
        {
            IsPopup = true;

            // Create our menu entries.
            MenuEntry startGameMenuEntry = new MenuEntry("Play");
            MenuEntry exitMenuEntry = new MenuEntry("Exit");

            // Hook up menu event handlers.
            startGameMenuEntry.Selected += StartGameMenuEntrySelected;
            exitMenuEntry.Selected += OnCancel;

            // Add entries to the menu.
            MenuEntries.Add(startGameMenuEntry);
            MenuEntries.Add(exitMenuEntry);
        }

        // Handles "Play" menu item selection
        void StartGameMenuEntrySelected(object sender, EventArgs e)
        {
            ScreenManager.AddScreen(new InstructionsScreen(), null);
        }

        // Handles "Exit" menu item selection		
        protected override void OnCancel(PlayerIndex playerIndex)
        {
            ScreenManager.Game.Exit();
        }

        protected override void UpdateMenuEntryLocations()
        {
            base.UpdateMenuEntryLocations();

            foreach (var entry in MenuEntries)
            {
                var position = entry.Position;

                position.Y += 60;

                entry.Position = position;
            }
        }
    }
}
